packages/grafana-data/src/transformations/transformers/fillGaps.test.ts

```diff
@@ -0,0 +1,71 @@
+import { FieldType, MutableDataFrame } from '@grafana/data';
+import { fillGaps } from './fillGaps';
+
+describe('fillGaps Transformer', () => {
+  test('should fill nulls as expected', () => {
+    const df = new MutableDataFrame({
+      refId: 'A',
+      fields: [
+        { name: 'Time', type: FieldType.time, values: [1, 3, 10] },
+        { name: 'One', type: FieldType.number, config: { interval: 1 }, values: [4, 6, 8] },
+        { name: 'Two', type: FieldType.string, values: ['a', 'b', 'c'] },
+      ],
+    });
+    const result = fillGaps(df);
+    expect(result.fields[0].values.toArray()).toStrictEqual([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+    expect(result.fields[1].values.toArray()).toStrictEqual([4, null, 6, null, null, null, null, null, null, 8]);
+    expect(result.fields[2].values.toArray()).toStrictEqual(['a', null, 'b', null, null, null, null, null, null, 'c']);
+  });
+
+  test('should not insert nulls for out of order timestamps', () => {
```

Comment:
i'm pretty sure we pre-sort timestamp fields when needed, so we can save some cycles by not handling this, perhaps.

Reply:
yeah, the transformation doesn't handle this case specifically. the test was written before the code, so the test is just there to make sure the behavior doesn't change